src/lib/structures/symbols.ml

```diff
@@ -317,82 +296,242 @@ let name_to_string =
+    | F_Clause _ -> Fmt.pf ppf "or"
+    | F_Lemma -> failwith "cannot print lemma in the SMT-LIB format"
+    | F_Skolem -> failwith "cannot print skolem function in the SMT-LIB format"
+    | F_Iff -> Fmt.pf ppf "="
+    | F_Xor -> Fmt.pf ppf "xor"
+
+  let pp ppf sy =
+    match sy with
+    (* Core theory *)
+    | True -> Fmt.pf ppf "true"
+    | False -> Fmt.pf ppf "false"
+    | Name (n, _, _) -> pp_name ppf (Hstring.view n)
+    | Var v -> Fmt.string ppf (Var.to_string v)
+    | Op Tite -> Fmt.pf ppf "ite"
+
+    (* Reals and Ints theories *)
+    | Int i ->
+      if Z.sign i = -1 then
+        Fmt.pf ppf "(- %a)" Z.pp_print (Z.abs i)
+      else
+        Fmt.pf ppf "%a" Z.pp_print i
+    | Real q ->
+      if Z.equal (Q.den q) Z.one then
+        Fmt.pf ppf "%a.0" Z.pp_print (Q.num q)
+      else if Q.sign q = -1 then
+        Fmt.pf ppf "(/ (- %a) %a)"
+          Z.pp_print (Z.abs (Q.num q))
+          Z.pp_print (Q.den q)
+      else
+        Fmt.pf ppf "(/ %a %a)" Z.pp_print (Q.num q) Z.pp_print (Q.den q)
+    | Op Plus -> Fmt.pf ppf "+"
+    | Op Minus -> Fmt.pf ppf "-"
+    | Op Mult -> Fmt.pf ppf "*"
+    | Op Div -> Fmt.pf ppf "/"
+    | Op Modulo -> Fmt.pf ppf "%%"
+
+    (* Reals_Ints theory *)
+    | Op Abs_int | Op Abs_real -> Fmt.pf ppf "abs"
+    | Op Real_of_int -> Fmt.pf ppf "to_real"
+    | Op Real_is_int -> Fmt.pf ppf "is_int"
+
+    (* FixedSizedBitVectors theory *)
+    | Bitv (n, s) ->
+      Fmt.pf ppf "#b%s" (Z.format (Fmt.str "%%0%db" n) s)
+    | Op Extract (i, j) ->
+      Fmt.pf ppf "(_ extract %d %d)" i j
+    | Op Concat -> Fmt.pf ppf "concat"
+    | Op BV2Nat -> Fmt.pf ppf "bv2nat"
+    | Op BVnot -> Fmt.pf ppf "bvnot"
+    | Op BVand -> Fmt.pf ppf "bvand"
+    | Op BVor -> Fmt.pf ppf "bvor"
+
+    (* ArraysEx theory *)
+    | Op Get -> Fmt.pf ppf "select"
+    | Op Set -> Fmt.pf ppf "store"
+
+    (* DT theory *)
+    | Op Record -> ()
+    | Op (Access s | Constr s | Destruct (s, _)) -> Hstring.print ppf s
+
+    | Form f -> pp_form ppf f
+    | Lit lit -> pp_lit ppf lit
+    | Let -> Fmt.pf ppf "let"
+
+    (* Not in the SMT-LIB standard *)
+    | Op (Int2BV _) -> Fmt.pf ppf "(_ int2bv)"
+    | Op Not_theory_constant -> Fmt.pf ppf "(_ not_theory_constant)"
+    | Op Is_theory_constant -> Fmt.pf ppf "(_ is_theory_constant)"
+    | Op Linear_dependency -> Fmt.pf ppf "(_ linear_dependency)"
+    | Op Sqrt_real -> Fmt.pf ppf "(_ sqrt_real)"
+    | Op Sqrt_real_default -> Fmt.pf ppf "(_ sqrt_real_default)"
+    | Op Sqrt_real_excess -> Fmt.pf ppf "(_ sqrt_real_excess)"
+    | Op Int_floor -> Fmt.pf ppf "(_ int_floor)"
+    | Op Int_ceil -> Fmt.pf ppf "(_ int_ceil)"
+    | Op Max_real -> Fmt.pf ppf "(_ max_real)"
+    | Op Max_int -> Fmt.pf ppf "(_ max_int)"
+    | Op Min_real -> Fmt.pf ppf "(_ min_real)"
+    | Op Min_int -> Fmt.pf ppf "(_ min_int)"
+    | Op Integer_log2 -> Fmt.pf ppf "(_ integer_log2)"
+    | Op Pow -> Fmt.pf ppf "(_ pow)"
+    | Op Integer_round -> Fmt.pf ppf "(_ integer_round)"
```

Comment:
These should not be indexed (there should be no `_`).

Reply:
Well except for `Int2BV` which should be `(_ int2bv n)` for `Int2BV n`.

Reply:
`Integer_round` is not `ae.round`. `Integer_round` does not exist in the SMT-LIB format. `ae.round` is `Float`.

Reply:
Ah but yes for custom primitives we should prefix with `ae` that is a good idea! BTW, `Int_floor` is `to_int` from the Reals_Ints theory.